docs/development/Offline_mode.md

```diff
@@ -76,3 +76,130 @@ The main problem is deciding what changes should be saved and what changes shoul
+
+### NoSQL / Storing plain JSON
+Our first approach was to store the string value of `to_json()` of each model in a local database. With this, you could get all models back from the database with their `from_json()` method. However, this raises a lot of errors when dealing with relational models. Consider this example:
+
+A model `A` has a one-to-many with the model `B`. If an instance of `B`, related to an instance `A`, was to be altered, it would not be altered on `A`.
+
+Thus, storing the plain `to_json()` string values from models is not a valid solution on related models.
+
+
+### In-memory design
+Since mobile devices has become more powerfull and reliable, an in-memory approach is strongly considerable. All models needed for the repository is provided by the api_client making the solution slightly straight forward and performance-efficient (https://en.wikipedia.org/wiki/In-memory_database). 
+
+However, this solution depends on the use case. If the app crashes, closes or the device is turned off obviously all data is lost.
+
+### The 1-1 approach
+Another very straight forward solution is a 1-1 relational database design with the web-api. This requires the offline repository to have the exact same rows, columns and tables as the web-api.
+
+Be careful with this solution. It seems benefiting, but do you really want to implement yet another thing for future students to maintain? If something is to be changed in the model layer, this will be your workload:
+
+1. Alter the modellayer in the web-api
+2. Migrate the database
+3. Customize the unittests
+4. Customize the integration tests
+5. Alter the models in the api_client
+6. Customize the unittests
+7. Integration test between web_api and api_client
+8. Alter the offline repository
+9. Alter the unittests
+10. Integration test between the offline repository and the api_client
+11. Integration test between the offline repository and the web-api
+12. Alter the weekplanner to use the new feature
+13. Integration test between weekplanner and the offline repository
+
+And do not forget that this will require 4 pull requests to 4 different projects. How will you convince your code reviewers that it works?
+
+Our next approach tries to avoid this problem.
+
+### A better json storage solution
+From our point of view, a more complex form of "json"-storage is needed. This is our recommendation:
+
+*This is an abstract description and will not describe how to actual implement it*.
+
+Assume that the `to_json()` and `from_json()` is bijective and all objects contains something unique (the identifier from the web-api would be essential), then you could:
+
+1. Create all database tables at runtime (when you go offline) from the models `to_json` or `from_json` method

```

Comment:
Not sure how this would work. When offline, you cannot access the online database which is what these methods do from my understanding.

Reply:
Well, off course the workflow would be
1. The user hits a button "goes offline"
2. Build the database tables
3. Actually go offline

Reply:
Okay, I thought this was against the wishes of the stakeholders but maybe I'm wrong. My understanding was that it should all happen automatically so that the users should never have to think about it. In any case, this workflow should probably be more explicit

Reply:
Maybe the "Create all database tables at runtime" should be removed, since this my not be how one would do it